nix-darwin/default.nix

```diff
@@ -9,7 +9,11 @@ let
   extendedLib = import ../modules/lib/stdlib-extended.nix pkgs.lib;
 
   hmModule = types.submoduleWith {
-    specialArgs = { lib = extendedLib; };
+    specialArgs = { 
+      lib = extendedLib; 
+      darwinConfig = config;
+    };
+    } // cfg.extraSpecialArgs;
```

Comment:
Maybe this line is intended for the next commit?

Reply:
Ohh right it was a hiccup while trying to split up the changes with `git add --patch`. Just fixed, thanks for the catch!